Задача: Секреты дроидов.

piligrim-curious opened the pull request:
**Droid**: Класс дроидов.
**Main**: Демонстрация работы;
**DroidMessageEncryptor**: Функциональный интерфейс.

piligrim-curious committed:
Задача: Секреты дроидов.
Droid: Класс дроидов.
Main: Демонстрация работы;
DroidMessageEncryptor: Функциональный интерфейс.

SnurnicynKonstantin commented on src/main/java/school/faang/secrets/droids/Droid.java:
```diff
@@ -0,0 +1,48 @@
+        return encryptor.encrypt(message, key);
+    }
+
+    String decryptMessage(String message, int key) {
```
Оба метода очень похоже друг на друга. Можно в один сжать. Например.
Для шифрования:
` encryptMessage(encryptedMessage, encryptionKey);`
 Для дешифрования:
`encryptMessage(encryptedMessage, (-1) * encryptionKey + 26);`

-1 и 26 в константы конечно

SnurnicynKonstantin replied:
А еще лучше енкриптор в константу вынести:
```
private static final  DroidMessageEncryptor ENCRYPTOR = (msg, encryptionKey) -> {
            StringBuilder encryptedMessage = new StringBuilder();
            for (char ch : msg.toCharArray()) {
                if (Character.isLetter(ch)) {
                    char base = Character.isLowerCase(ch) ? 'a' : 'A';
                    encryptedMessage.append((char) ((ch - base + encryptionKey) % 26 + base));
                } else {
                    encryptedMessage.append(ch);
                }
            }
            return encryptedMessage.toString();
        };
```

Тогда можно будет два коротких метода сделать:
```
    private String encryptMessage(String message, int key) {
        return ENCRYPTOR.encrypt(message, key);
    }

    private String decryptMessage(String encryptedMessage, int key) {
        return ENCRYPTOR.encrypt(encryptedMessage, ALPHABET_LENGTH - key);
    }
```

piligrim-curious committed:
Задача: Секреты дроидов.
Droid: Класс дроидов.
Main: Демонстрация работы;
DroidMessageEncryptor: Функциональный интерфейс.

Изменения:
(Droid.java)
* Изменена логика шифрования/дешифрования.

SnurnicynKonstantin reviewed (approved):
Теперь совсем красиво получилось. И в один метод уложился и енкриптор вынес. Молдец!
